src/tools/cgsnapshot.c

```diff
@@ -767,20 +767,24 @@ int main(int argc, char *argv[])
 		output_f = stdout;
 
 	/* blacklkist */
-	if (flags & FL_BLACK) {
+	if (flags & FL_BLACK)
 		ret  = load_list(bl_file, &black_list);
-	} else {
+	else {
 		/* load the blacklist from the default location */
 		ret  = load_list(BLACKLIST_CONF, &black_list);
 	}
-	if (ret != 0)
+	if (ret != 0) {
+		ret = ret ? EXIT_BADARGS : 0;
```

Comment:
Won't this always evaluate to `EXIT_BAD_ARGS` since in the `if` line above we checked if `ret != 0`

Reply:
Good catch :). Sure will change it here and other places too.